lib/server/routes/buckets.js

```diff
@@ -134,26 +134,14 @@ BucketsRouter.prototype._validate = function (req, res, next) {
  * @param {http.ServerResponse} res
  * @param {Function} next
  */
-BucketsRouter.prototype.getBuckets = function (req, res, next) {
-  let findQuery = { userId: req.user.uuid };
+BucketsRouter.prototype.getBuckets = async function (req, res, next) {
+  const userId = req.user.uuid;
   const startDate = utils.parseTimestamp(req.query.startDate);
-  if (startDate) {
-    findQuery.created = { $gt: startDate };
-  }
 
-  this.storage.models.Bucket
-    .find(findQuery)
-    .sort({ created: 1 })
-    .limit(constants.DEFAULT_MAX_BUCKETS)
-    .exec(function (err, buckets) {
-      if (err) {
-        return next(new errors.InternalError(err.message));
-      }
+  const buckets = await this.usecase.findAllByUserAndCreatedSince(userId, startDate)
+    .catch(err=> next(new errors.InternalError(err.message)));
```

Comment:
As this catch resolves, the code will continue, using res.send twice and triggering a Node.js error. Use a try catch to prevent that.